### PR TITLE
fix(fec-7334): the player state has not saved on change media

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes
on change media we are considering the player config but not to the current state of the following player attributes: 
* muted
* volume
* text language
* audio language
* playback rate

the solution is: 
* saving the state of the attributes above
* considering them (on change media) before player config  
 
### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1002,53 +1019,71 @@ export default class Player extends FakeEventTarget {
...
+        this._playbackAttributesState.TEXT_LANG = event.payload.selectedTextTrack.language;
         this._markActiveTrack(event.payload.selectedTextTrack);
         return this.dispatchEvent(event);
       });
       this._eventManager.listen(this._engine, CustomEvents.TEXT_CUE_CHANGED, (event: FakeEvent) => this._onCueChange(event));
       this._eventManager.listen(this._engine, CustomEvents.ABR_MODE_CHANGED, (event: FakeEvent) => this.dispatchEvent(event));
       this._eventManager.listen(this, Html5Events.PLAY, this._onPlay.bind(this));
       this._eventManager.listen(this, Html5Events.ENDED, this._onEnded.bind(this));
+      this._eventManager.listen(this, Html5Events.VOLUME_CHANGE, () => {
+        this._playbackAttributesState.MUTED = this.muted;
```
Comment (dan-ziv) [inline src/player.js]: split to MUTE_CHANGE

File: src/player.js [hunk 2]
```diff
@@ -1104,6 +1139,9 @@ export default class Player extends FakeEventTarget {
       this._firstPlay = false;
       this.dispatchEvent(new FakeEvent(CustomEvents.FIRST_PLAY));
       this._posterManager.hide();
+      if (typeof this._playbackAttributesState.RATE === 'number') {
+        this.playbackRate = this._playbackAttributesState.RATE;
```
Comment (dan-ziv) [inline src/player.js]: place the update for playbackRate in ratechange event listener (need to add if we don't have this event)

Comment (yairans) [reply src/player.js]: @dan-ziv 
in `ratechange` i'm saving the rate value. here i'm using it.

File: src/player.js (hunk 1 above)
Comment (yairans) [reply src/player.js]: done

File: src/player.js (hunk 2 above)
Comment (dan-ziv) [reply src/player.js]: So why here?
We cannot assign it also in `_handlePlaybackOptions`?

Comment (yairans) [reply src/player.js]: no. you can change the rate only after loading/play

File: src/player.js [hunk 3]
```diff
@@ -1270,8 +1310,8 @@ export default class Player extends FakeEventTarget {
 
     this.hideTextTrack();
 
-    this._setDefaultTrack(TrackTypes.TEXT, this._getLanguage(playbackConfig.textLanguage, activeTracks.text, TrackTypes.TEXT), offTextTrack);
-    this._setDefaultTrack(TrackTypes.AUDIO, playbackConfig.audioLanguage, activeTracks.audio);
+    this._setDefaultTrack(TrackTypes.TEXT, this._playbackAttributesState.TEXT_LANG, this._getLanguage(playbackConfig.textLanguage, activeTracks.text, TrackTypes.TEXT), offTextTrack);
```
Comment (OrenMe) [inline src/player.js]: @yairans `playbackConfig.textLanguage` is initial config and if there is `this._playbackAttributesState.TEXT_LANG` already it means that it replaced the initial config, so why pass both now? just pass this or that to the `_setDefaultTrack` method

File: src/player.js [hunk 4]
```diff
@@ -226,6 +226,19 @@ export default class Player extends FakeEventTarget {
    * @private
    */
   _streamType: string;
+  /**
+   * The current playback attributes state
+   * @type {Object}
+   * @private
+   */
+  _playbackAttributesState: { [attribute: string]: any } = {
+    MUTED: undefined,
```
Comment (OrenMe) [inline src/player.js]: @yairans why use CAPITAL? it's not a const just an object map. please use lower case.

File: src/player.js (hunk 3 above)
Comment (yairans) [reply src/player.js]: @OrenMe the `this._playbackAttributesState.TEXT_LANG` could be non exists in the new media. for this case we have to take the lang form config.

Comment (OrenMe) [reply src/player.js]: The original config can be non exist as well, it doesn’t matter let this case. The point is, the only way you have a config here is if it was previously set, but is is only one, either from app config, auto selected or user selected.

File: src/player.js (hunk 4 above)
Comment (yairans) [reply src/player.js]: done

**Pull request merged**